src/smart-components/role/RolesTable.tsx

```diff
@@ -81,12 +94,31 @@ const RolesTable: React.FunctionComponent<RolesTableProps> = ({ selectedRole })
     fetchData({
       limit: perPage,
       offset: (page - 1) * perPage,
-      orderBy: 'display_name',
+      orderBy: `${direction === 'desc' ? '-' : ''}${sortBy}` || '',
       count: totalCount || 0,
+      filters: filters,
     });
-  }, [fetchData, page, perPage]);
+  }, [fetchData, page, perPage, filters, onSetFilters, sortBy, direction]);
```

Comment:
@InsaneZein one last comment - it would be better to use a separate useEffect for filtering and add some debounce - otherwise, it shoots a lot of requests as you are typing